### ISSUE TITLE
The training did not report any errors, but it seemed like it had stopped, but the GPU was still in use

### ISSUE THREAD

**Issue by nicekwow** [open-mmlab/mmsegmentation#3403]
Thanks for your error report and we appreciate it a lot.

**Checklist**

1. I have searched related issues but cannot get the expected help.
2. The bug has not been fixed in the latest version.

**Describe the bug**

![image](https://github.com/open-mmlab/mmsegmentation/assets/93817732/5715f2ec-04a3-4e93-985b-34311a99db88)
The training did not report any errors, but it seemed like it had stopped, but the GPU was still in use
![image](https://github.com/open-mmlab/mmsegmentation/assets/93817732/eb9603ba-2fac-4c25-81ab-fac3e42b6de6)

**Reproduction**

1. What command or script did you run?

   ```none
   python tools/train.py configs/mae/mae-base_upernet_8xb2-amp-160k_ade20k-512x512_mydata.py
   ```

2. Did you make any modifications on the code or config? Did you understand what you have modified?
I used my own dataset for training and loaded the pre training weights obtained from mmpretrain

4. What dataset did you use?
    my own dataset，like this
![image](https://github.com/open-mmlab/mmsegmentation/assets/93817732/091774cf-9b41-43d0-83aa-3f59d771db72)
There is such content in both train and val
![image](https://github.com/open-mmlab/mmsegmentation/assets/93817732/de713ece-cb14-48ea-a617-057dfbf84957)

**Environment**

sys.platform: win32
Python: 3.8.0 (default, Nov  6 2019, 16:00:02) [MSC v.1916 64 bit (AMD64)]
CUDA available: True
numpy_random_seed: 2147483648
GPU 0: NVIDIA GeForce RTX 3080
CUDA_HOME: C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.3\bin\nvcc.exe
C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v9.0\bin\nvcc.exe
C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.4
MSVC: 用于 x64 的 Microsoft (R) C/C++ 优化编译器 19.29.30145 版
GCC: n/a
PyTorch: 1.12.1
PyTorch compiling details: PyTorch built with:
  - C++ Version: 199711
  - MSVC 192829337
  - Intel(R) Math Kernel Library Version 2020.0.2 Product Build 20200624 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v2.6.0 (Git Hash 52b5f107dd9cf10910aaa19cb47f3abf9b349815)
  - OpenMP 2019
  - LAPACK is enabled (usually provided by MKL)
  - CPU capability usage: AVX2
  - CUDA Runtime 11.3
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_61,code=sm_61;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_80,code=sm_80;-gencode;arch=compute_86,code=sm_86;-gencode;arch=compute_37,code=compute_37
  - CuDNN 8.3.2  (built against CUDA 11.5)
  - Magma 2.5.4
  - Build settings: BLAS_INFO=mkl, BUILD_TYPE=Release, CUDA_VERSION=11.3, CUDNN_VERSION=8.3.2, CXX_COMPILER=C:/cb/pytorch_1000000000000/work/tmp_bin/sccache-cl.exe, CXX_FLAGS=/DWIN32 /D_WINDOWS /GR /EHsc /w /bigobj -DUSE_PTHREADPOOL -openmp:experimental -IC:/cb/pytorch_1000000000000/work/mkl/include -DNDEBUG -DUSE_KINETO -DLIBKINETO_NOCUPTI -DUSE_FBGEMM -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -DEDGE_PROFILER_USE_KINETO, LAPACK_INFO=mkl, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, TORCH_VERSION=1.12.1, USE_CUDA=ON, USE_CUDNN=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=OFF, USE_MPI=OFF, USE_NCCL=OFF, USE_NNPACK=OFF, USE_OPENMP=ON, USE_ROCM=OFF,

TorchVision: 0.13.1
OpenCV: 4.8.1
MMEngine: 0.9.0
MMSegmentation: 1.2.1+cbf9af1

**Comment by lxy-mini** [comment]
You can check your data format. I used rgb label and had the same problem as you.
